src/api/query-keys.ts

```diff
@@ -0,0 +1,37 @@
+export const queryKeys = {
+  camps: {
+    all: ['camps'] as const,
+  },
+  staffs: {
+    all: ['staffs'] as const,
+  },
+  users: {
+    all: ['users'] as const,
+    me: () => [...queryKeys.users.all, 'me'] as const,
+    detail: (userId: string) => [...queryKeys.users.all, userId] as const,
+  },
+  events: {
+    all: (campId: number | undefined) => ['events', campId] as const,
```

Comment:
nitpick: 好みの問題ではありますが、`?`を使ったほうが短くて良いと思います（どっちの書き方でも型は同じなので本当にどっちでも良いんですが、混在しているのがちょっと気になったので一応）
```suggestion
    all: (campId?: number) => ['events', campId] as const,
```